### ISSUE TITLE
Consider making some of the data generation function public

### ISSUE THREAD

**Issue by Remi-Gau** [nilearn/nilearn#4960]
Going thorough the examples I see this

```
from nilearn._utils.data_gen import generate_group_sparse_gaussian_graphs
```

https://nilearn.github.io/dev/auto_examples/03_connectivity/plot_simulated_connectome.html#connectivity-structure-estimation-on-simulated-data

I'd be tempted to say that it my be useful for some other examples and bug reports to be able to use on synthetic data:
- no need to download data
- more flexibility (data dimension...)

**Comment by bthirion** [comment]
Yes, but the downside is that it makes the examples more toyish and thus may reduce user's confidence...
So we need to find a balance in that respect.

**Comment by man-shu** [comment]
Having both generated and real datasets can be good. 
How sklearn does it: https://scikit-learn.org/stable/datasets.html
